library/std/src/sys/thread_local/static_local.rs

```diff
@@ -25,7 +23,8 @@ pub macro thread_local_inner {
             // FIXME(#84224) this should come after the `target_thread_local`
             // block.
             static mut VAL: $t = INIT_EXPR;
-            unsafe { $crate::option::Option::Some(&VAL) }
+            // SAFETY: we only ever create shared references, so there's no mutable aliasing.
+            unsafe { $crate::option::Option::Some(&*addr_of!(VAL)) }
```

Comment:
This macro is never imported:
```suggestion
            unsafe { $crate::option::Option::Some(&*$crate::ptr::addr_of!(VAL)) }
```

Reply:
Ah, I guess this does not actually get built by PR CI...